tests/integration/components/country-chooser-test.js

```diff
@@ -0,0 +1,177 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, setupOnerror } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+import { clickTrigger, selectChoose } from 'ember-power-select/test-support/helpers'
+
+import CountryCodes from '@upfluence/ember-upf-utils/resources/country-codes';
+
+const DROPDOWN_CLASS = '.item-chooser .ember-basic-dropdown';
+
+module('Integration | Component | country-chooser', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it throws an error if no onCountrySelection action is passed', async function(assert) {
+    setupOnerror((err) => {
+      assert.equal(
+        err.message,
+        '[component][country-chooser] Please provide a onCountrySelection action',
+        'Throws as an error about missing onCountrySelection action.'
+      );
+    });
+
+    await render(hbs`<CountryChooser />`);
+  });
+
+  test('it displays all of the available countries as options', async function(assert) {
+    this.onCountrySelection = () => {};
+
+    await render(hbs`<CountryChooser @onCountrySelection={{this.onCountrySelection}} />`);
+
+    assert.dom(DROPDOWN_CLASS).exists();
+
+    await clickTrigger();
+
+    const options = this.element.querySelectorAll('.ember-power-select-options li');
+
+    assert.equal(options.length, CountryCodes.length);
+
+    let optionsCountries = [];
+    options.forEach((v) => optionsCountries.push(v.innerHTML.trim()))
```

Comment:
Why this ? Because `querySelectorAll` returns a `NodeList` instance, which his iterable but has no `map` method. So no  `map((o) => o.innerHTML.trim()` possible 😢